sdk/java/pulumi/src/main/java/com/pulumi/core/internal/Codegen.java

```diff
@@ -53,6 +54,10 @@ public static <T, O extends Output<T>> Output<T> ofNullable(@Nullable O value) {
         return value == null ? Codegen.empty() : value;
     }
 
+    public static Config config(String name) {
```

Comment:
to make sure we cleanly separate the internal codegen API, the config factory method should be moved here IMHO

changes to the codegen are out of scope for this PR